### ISSUE TITLE
[Server] Implement LeaderBoard Feature to each Quiz.

### ISSUE THREAD

**Issue by shubhansu-kr** [mvp-2003/Quikz#23]
## Leaderboard
**Description:**  
Create a leaderboard to show top-performing users based on quiz results.

**Tasks:**
- Create leaderboard model.
- Implement routes to fetch leaderboard data.
- Update leaderboard based on quiz results and user performance.

**Dependencies:**  
- **Quiz Management** (Issue 2) to collect quiz scores.
  
**Guidelines for Assignee:**
- Comment on the issue with the model and routes for the leaderboard feature.

**Comment by Alienimnida** [comment]
Hello there! Can I take this up if it is not assigned to anyone right now?

**Comment by shubhansu-kr** [comment]
HI @Alienimnida ,  You can start working on this issue. Please share your approach here before starting the implementation part. Feel free to reach out if you need more info. Thanks.

**Comment by Alienimnida** [comment]
I'll let you know as soon as I set it up in my system

**Comment by shubhansu-kr** [comment]
Any update @Alienimnida

**Comment by shubhansu-kr** [comment]
No issues if you need time. I was just confirming if you are still working in case if you are not i would assign to someone else. 

Glad you are still contributing, Thanks

**Comment by Alienimnida** [comment]
Yes I really want to work on it I just need some time actually this Hacktoberfest usually coincide with our festive season in Kolkata but I'll get it done ASAP

**Comment by Alienimnida** [comment]
Hello I was setting up the project in my machine, the landing page is appearing but when I am clicking on join a quiz or create a quiz none of the routes are working instead it is throwing a 404 error. In the backend I have included the MONGO_URI the backend is working fine.

**Comment by Alienimnida** [comment]
Also could you please change Open your browser and go to http://localhost:3000/ to see Qwikz in action. this part in read me file as the application is working on port 5173.

**Comment by Alienimnida** [comment]
Could you also please tell me how can I get the Quiz Scores from (Issue 2) like you have mentioned above?

**Comment by shubhansu-kr** [comment]
> Hello I was setting up the project in my machine, the landing page is appearing but when I am clicking on join a quiz or create a quiz none of the routes are working instead it is throwing a 404 error. In the backend I have included the MONGO_URI the backend is working fine.

That must be because we havent integrated the frontend with backend

**Comment by shubhansu-kr** [comment]
> Also could you please change Open your browser and go to http://localhost:3000/ to see Qwikz in action. this part in read me file as the application is working on port 5173.

You can update that as a part of this PR

**Comment by shubhansu-kr** [comment]
> Could you also please tell me how can I get the Quiz Scores from (Issue 2) like you have mentioned above?

I guess you would have to incorporate that. Let me check.

**Comment by Alienimnida** [comment]
Ok I will update that as part of PR and should I integrate the frontend with the backend myself?

**Comment by shubhansu-kr** [comment]
Please create an issue for it. I'll assign it to you.

**Comment by Alienimnida** [comment]
Hello I'm sorry to inform you that I won't be able to continue please unassign me from all the issues. Once again extremely sorry for wasting everyone's time.

**Comment by shubhansu-kr** [comment]
No worries and No need to apologise. Thanks for your contribution so far. 

You can un assign yourself.

**Comment by Alienimnida** [comment]
Thank you very much and I am extremely sorry that I couldn't fullfil the commitment.
